server/app/services/applicant/question/NumberQuestion.java

```diff
@@ -35,17 +35,23 @@ public ImmutableList<Path> getAllPaths() {
 
   @Override
   protected ImmutableMap<Path, ImmutableSet<ValidationErrorMessage>> getValidationErrorsInternal() {
-    // TODO(#1944): Move this validation from the top-level to a field-specific error.
-    // Presently, the renderer displays a hidden error when the input can't be converted
-    // to the correct currency.
-    return ImmutableMap.of(applicantQuestion.getContextualizedPath(), validateNumber());
+    return ImmutableMap.of(getNumberPath(), validateNumber());
   }
 
   private ImmutableSet<ValidationErrorMessage> validateNumber() {
+    Optional<Long> numberValue = getNumberValue();
+    // When staging updates, the attempt to update ApplicantData would have failed to
+    // convert to a number and been noted as a failed update. We check for that here.
+    if (applicantQuestion.getApplicantData().getFailedUpdates().containsKey(getNumberPath())
```

Comment:
suggestion (blocking): may as well encapsulate this logic for better law of demeter adherence:
```java
class ApplicantData {
  public boolean updateDidFailAt(Path path) {
    return getFailedUpdates().containsKey(path);
  }
}
```

Reply:
Done.